### ISSUE TITLE
Offer Out Of Gas

### ISSUE THREAD

**Issue by micahalcorn** [OriginProtocol/origin#3400]
Using MetaMask on desktop just now with the "fast" gas setting 🦊

> [Etherscan Link](https://etherscan.io/tx/0xaaac75272f828bd4464b872be5a824116e97eabea1371826593edb4249b36d4a)

How can I move the funds from my proxy to the marketplace contract?

<img width="634" alt="Screen Shot 2019-09-14 at 5 59 44 PM" src="https://user-images.githubusercontent.com/273937/64915374-f53bc500-d719-11e9-9aff-68b840f6d40a.png">

<img width="1440" alt="Screen Shot 2019-09-14 at 5 59 15 PM" src="https://user-images.githubusercontent.com/273937/64915378-02f14a80-d71a-11e9-8fb6-3d5f09748691.png">

-----

At Origin, we use [ZenHub](https://www.zenhub.com/) to manage our engineering tasks and product development. [Download their browser extension](https://www.zenhub.com/extension) and check out our open workspace at [github.com/originprotocol/origin#zenhub](https://github.com/originprotocol/origin#zenhub).

**Comment by franckc** [comment]
@micahalcorn gasLimit is supposed to be set to 1M when making an offer via a proxy ([see code](https://github.com/OriginProtocol/origin/blob/master/packages/graphql/src/mutations/_txHelper.js#L220)). But your tx shows gasLimit set to 100k... Did you maybe by mistake modify it in MetaMask before submitting ?

**Comment by micahalcorn** [comment]
Ah, sorry @franckc that was probably it. I thought I reset everything to defaults after I was trying to get cute with it. 😇 I suppose we don't really need to handle this case of it not reverting because most users won't mess with the configuration, and at least the funds are still in their control.

**Comment by franckc** [comment]
Phew. Glad it was just human error... ! Sorry about your funds stuck in proxy, we'll get them out of there, I promise...

**Comment by mikeshultz** [comment]
I don't think there's any way we can prevent users meddling with gas limit and price in some web3 browsers/wallets.  If the user does choose to alter these, its at their own risk.  And the risk here is only of moving funds to the proxy that they control.

I'd vote to close this.

**Comment by franckc** [comment]
@micahalcorn Ok if we close this issue ?

**Comment by micahalcorn** [comment]
I agree that we shouldn't make any dramatic changes, but I would think that we could recognize the error and fail more gracefully than we currently do. I'll deprioritize this and unpin it as high-priority. ⬇️

**Comment by micahalcorn** [comment]
I've renamed this issue and propose that we communicate the issue to the user instead of just punting to a nasty console. 🏈 Also, if possible, we should make a distinction between "out of gas" errors where the user screwed with the settings and those where we just made the wrong assumption.

**Comment by mikeshultz** [comment]
Well, in general the dapp needs better error handling and to actually notify the user of what happened.  Kind of part of the discussion going on in #3229
